modal/functions.py

```diff
@@ -355,6 +375,20 @@ async def get_all_outputs():
                     del pending_outputs[item.input_id]
                     yield item
 
+    async def get_all_outputs_and_clean_up():
+        try:
+            async for item in get_all_outputs():
+                yield item
+        finally:
+            # "ack" that we have all outputs we are interested in and let backend clear results
+            request = api_pb2.FunctionGetOutputsRequest(
```

Comment:
This feels a bit like "boolean flag" type code smell – where the function does two very different things based on whether the flag is present or not. I think it would be cleaner if it is a separate method?

Reply:
It might look like it's a flag to make the call into an "ack", but this *extra* acking call is only performed for maps and generators. The flag is an "actual flag" for the single-input use case, where there is no ack-call and everything is done in the same call. Does that make sense or do you think it's still a boolean flag type code smell?
Perhaps the flag is badly named? Maybe it should be named "final_request" or something, but it could be misleading as it only has effect if an actual output is returned (if not, further requests will be allowed)